docroot/modules/custom/foia_users/foia_users.module

```diff
@@ -40,35 +35,57 @@ function _office_user_nid_access($node, $account) {
 function foia_users_node_access($node, $op, $account) {
 
   // Restrict editing node based on matching of 'field_agency' entity reference.
-  if ($node->bundle() == 'foia_office') {
-
+  if ($node->bundle() == 'agency_component') {
+    // Get the current user's roles.
     $user = User::load(\Drupal::currentUser()->id());
     $user_roles = $user->getRoles();
 
-    if (in_array("foia_office_creator", $user_roles) || in_array("foia_office_reviewer", $user_roles)) {
+    if (in_array("agency_manager", $user_roles)) {
+      switch ($op) {
+        case 'update':
+          // Check if the user has the proper Agency association.
+          return _office_user_nid_access($node, $account);
```

Comment:
Same comment as above, we should rename this function to `_agency_user_nid_access`